### PR TITLE
[gh-2385] add dry run check for run_function.rs.

### LINKED ISSUE #2385: Do the dry run when executing normal transaction
When using `rooch move run` to execute a transaction, a dry run checks if the transaction can be executed properly.

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

Add dry run check for `run_function.rs`.

- Closes #2385

### CODE REVIEW

File: crates/rooch/build.rs [hunk 1]
```diff
@@ -11,5 +13,11 @@ fn main() -> Result<()> {
...
         .add_instructions(&RustcBuilder::all_rustc()?)?
         .emit()?;
+    read_proxy_from_env();
     Ok(())
 }
+
+fn read_proxy_from_env() {
+    println!("cargo:rerun-if-env-changed=https_proxy");
+    println!("cargo:rerun-if-env-changed=all_proxy");
+}
```
Comment (jolestar) [inline crates/rooch/build.rs:23]: Not about this feature.

File: crates/rooch/src/commands/rpc/commands/request.rs [hunk 2]
```diff
@@ -36,6 +36,11 @@ pub struct RequestCommand {
 #[async_trait]
 impl CommandAction<serde_json::Value> for RequestCommand {
     async fn execute(self) -> RoochResult<serde_json::Value> {
+        let https_proxy = env::var("https_proxy").ok();
```
Comment (jolestar) [inline crates/rooch/src/commands/rpc/commands/request.rs:39]: Not about this feature

Comment (feliciss) [reply crates/rooch/src/commands/rpc/commands/request.rs]: > Not about this feature

Sorry I committed wrong.

File: crates/rooch/build.rs (hunk 1 above)
Comment (feliciss) [reply crates/rooch/build.rs]: > Not about this feature.

Sorry I committed wrong.

File: crates/rooch/src/commands/move_cli/commands/run_function.rs [hunk 3]
```diff
@@ -81,15 +83,24 @@ impl CommandAction<ExecuteTransactionResponseView> for RunFunction {
             })
             .collect::<Result<Vec<_>>>()?;
         let action = MoveAction::new_function_call(function_id, type_args, args);
-        match (self.tx_options.authenticator, self.tx_options.session_key) {
+
+        let dry_run_result = context
```
Comment (jolestar) [inline crates/rooch/src/commands/move_cli/commands/run_function.rs:87]: If the `dry_run_result` is an error, we should return the result and not try to execute the tx.

File: crates/rooch/src/commands/move_cli/commands/run_function.rs [hunk 4]
```diff
@@ -147,10 +158,16 @@ impl CommandAction<ExecuteTransactionResponseView> for RunFunction {
...
                         .sign_and_execute(sender, action, Some(password), max_gas_amount)
-                        .await
+                        .await?
                 }
             }
+        };
+
+        if dry_run_result.raw_output.status != KeptVMStatusView::Executed {
+            result.error_info = Some(dry_run_result);
         }
```
Comment (jolestar) [inline crates/rooch/src/commands/move_cli/commands/run_function.rs:168]: Put this check before `execute(tx)`, or skip execute(tx) if dry_run_result is not KeptVMStatusView::Executed. 

If the dry run result is not Executed, we should not try to execute it.

The publish command also has this mistake. cc @steelgeek091

**Review by jolestar — APPROVED** [review]

**Comment by jolestar** [comment]
I merge this first.

File: crates/rooch/src/commands/move_cli/commands/run_function.rs (hunk 4 above)
Comment (feliciss) [reply crates/rooch/src/commands/move_cli/commands/run_function.rs:168]: > Put this check before `execute(tx)`, or skip execute(tx) if dry_run_result is not KeptVMStatusView::Executed.
> 
> If the dry run result is not Executed, we should not try to execute it.
> 
> The publish command also has this mistake. cc @steelgeek091

Shall I fix them together?

**Pull request merged**